parmed/openmm/parameters.py

```diff
@@ -458,11 +463,13 @@ def get_types(residue, atomname):
                 return [ a_types[a_names.index(atomname)] ]
 
         # Iterate over all residues
-        # TODO: Do we have to iterate over all patched residues too?
-        for name, residue in iteritems(self.residues):
+        for name, residue in chain(iteritems(self.residues), iteritems(self.patches)):
             for impr in residue._impr:
                 # Get the list of types involved in this improper
-                types = [ get_types(residue, atomname) for atomname in impr ]
+                try:
+                    types = [ get_types(residue, atomname) for atomname in impr ]
+                except:
```

Comment:
We should catch a narrower exception here.  What do we expect can be thrown?

Reply:
Fixed!